src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -1148,10 +1148,10 @@ private static IEnumerable<object[]> CustomInterfaces_Dictionaries()
         }
 
         [Fact]
-        public static void IReadOnlyDictionary_NonStringKey_NotSupported()
+        public static void IReadOnlyDictionary_NotSupportedKey()
         {
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<IReadOnlyDictionary<int, int>>(""));
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new GenericIReadOnlyDictionaryWrapper<int, int>()));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<IReadOnlyDictionary<Uri, int>>(@"{""http://foo"":1}"));
```

Comment:
Any particular reason to not support `Uri` (aside from not being in the spec)?

Reply:
No particular reason, when I was defining the [types supported](https://github.com/dotnet/runtime/pull/32676/files#diff-899ea4f41d4cbaeaa272936e99afcf9aR40) I thought of just include the ones that were supported by the Utf8Parser/Formatter + Enum. We can add Uri if you think there is value in supporting it.

Reply:
We can add this later if needed.